riot/riot.py

```diff
@@ -62,60 +66,155 @@ def to_list(x: t.Union[_K, t.List[_K]]) -> t.List[_K]:
 _T_stdio = t.Union[None, int, t.IO[t.Any]]
 
 
-@dataclasses.dataclass(unsafe_hash=True, eq=True)
+@dataclasses.dataclass(eq=True)
 class Interpreter:
-    _T_hint = t.Union[float, int, str]
+    _path: str = dataclasses.field()
+    _version: Version = dataclasses.field(init=False)
+    _version_expr = re.compile(f"python{VERSION_PATTERN}$", re.IGNORECASE | re.VERBOSE)
+
+    def __post_init__(self):
+        """Get the version of the interpreter."""
+        output = subprocess.check_output(
+            [self._path, "--version"], stderr=subprocess.STDOUT
```

Comment:
you might want to rebase on `master` we made some changes to how we get version from Python to alleviate the parsing of `--version` output.

https://github.com/DataDog/riot/blob/6d064caf94a51a4ce41600b1e0090004510a9567/riot/riot.py#L81-L91